### PR TITLE
refactor: rework project structure and routing logic

### PR DESCRIPTION
**Pull request by craigzour**
# Summary | Résumé

- Reworks project structure in both `src` and `tests` folders
- Reworks routing logic. All sub routers that were dispatched in multiple files have been regrouped in a single one.
- Refactors API endpoint handlers to be decoupled from Express framework. They now are `ApiOperations` and can be tested without `supertest`
- Removes legacy access token generation script. It is now replaced by API integration examples.
- Improves some unit tests by making sure audit logs are being generated when it is appropriate

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -22,9 +22,4 @@ REDIS_URL=redis://localhost:6379
 # This is required by the authentication feature but can be left empty if you are disabling it when doing local tests
 
 ZITADEL_DOMAIN=https://auth.forms-staging.cdssandbox.xyz
-ZITADEL_APPLICATION_KEY=YOUR_APPLICATION_KEY
-
-# (optional) Zitadel but only used by the utilTokenGeneration script to generate access token
-
-PROJECT_ID=123 # This is the GCForms project ID in Zitadel
```
Comment (craigzour) [inline .env.example:29]: This was for the `utilTokenGeneration.mjs` script that is being deleted with this PR

File: src/lib/connectors/freshdeskServiceConnector.ts
Comment (craigzour) [inline src/lib/connectors/freshdeskServiceConnector.ts:1]: Legacy code that had not been removed when the Freshdesk connector got refactored

File: src/middleware/authentication/middleware.ts [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
   next: NextFunction,
-) {
+): Promise<void> {
   const accessToken = request.headers.authorization?.split(" ")[1];
   const formId = request.params.formId;
 
   if (!accessToken) {
-    return response.sendStatus(401);
+    response.sendStatus(401);
+    return;
```
Comment (patheard) [inline src/middleware/authentication/middleware.ts:19]: Just curious why the change.  Not that I'm opposed, but was wondering.

Comment (craigzour) [reply src/middleware/authentication/middleware.ts:19]: Sure thing! With the introduction of the new `ApiOperation` pattern for implementing new API paths I had to create types that would conform to what Express expects. They need operation handlers to return either `void` or `Promise<void>` so the old `return response.something` was not working anymore.
Also when discussing this with Bryan, he mentioned that having the `return` on a different line also makes it clear that we are stopping the execution on purpose.

Comment (patheard) [reply src/middleware/authentication/middleware.ts:19]: Gotcha, that makes sense - thanks!

**Review by bryan-robitaille — APPROVED** [review]
